pkg/core/resource_graph.go

```diff
@@ -47,13 +50,30 @@ func (rg *ResourceGraph) AddResource(resource Resource) {
 //
 //	rg.AddDependency(lambda, lambda.Role)
 func (rg *ResourceGraph) AddDependency(deployedSecond Resource, deployedFirst Resource) {
+
 	for _, res := range []Resource{deployedSecond, deployedFirst} {
```

Comment:
Can't we just called rg.AddResource twice here? Are we thinking there will be more resources down the road?

Reply:
I originally wrote this - the adding used to be more complex. The "if not present" check used to be here, which is why I did the for loop. Once I moved that check into `AddResource`, I didn't bother to change this from a loop.

It should be two `AddResource` calls, but it's not anything Jordan changed.

Reply:
yeah i can go look at the code since i didnt change this method i need to take a peek, but if theres something we know needs changing, gordon maybe we can sync about it